lib/systems/default.nix

```diff
@@ -119,7 +119,14 @@ rec {
       # Just a guess, based on `system`
       inherit
         ({
-          linux-kernel = args.linux-kernel or {};
+          # The following per-cpu defaults `target` ensure that the
+          # image is stripped, which is now necessary in order to
+          # avoid circular dependencies.
+          linux-kernel = lib.optionalAttrs final.isMips {
+            target = "vmlinux.bin";
+          } // lib.optionalAttrs final.isPower {
+            target = "vmlinux.strip.gz";
+          } // (args.linux-kernel or {});
```

Comment:
Can you explain why this change is necessary? Doesn't platforms.nix already take care of it?  (Although I realise we're missing a big-endian POWER definition in platforms.nix.)

Reply:
It's in the commit message (copied below).  The upshot is that `platforms.nix` and `examples.nix` don't have "least upper bounds" -- there may be an assortment of `lib.systems.examples` entries which use CPU XYZ, but there isn't always (or even usually) some root `lib.systems.examples.xyz` which all of them extend from.

```
These defaults are placed within `lib.systems.default` rather than
`lib.systems.examples` since there isn't always a clear inheritance
hierarchy in the latter, let alone a "least upper bound of all
machines with CPU X".  The only other possible place where this
default could be placed would be in `pkgs/os-specific/linux/kernel`,
but at the moment that file isn't structured in a way that makes it
easy to achieve all three of the following goals:

1. Set a default value based on CPU type
2. Ensure that the default will be overridden by more-specific
   choices in `lib/systems/examples.nix`
3. Make it unlikely that refactorings of
   `pkgs/os-specific/linux/kernel` will accidentally lose the
   defaults.
```

Reply:
I had read the commit message but still didn't understand.



> there isn't always (or even usually) some root `lib.systems.examples.xyz` which all of them extend from.

Isn't this what the platforms.nix definitions are?

Reply:
No.  Maybe they should be, but they certainly aren't.